### PR TITLE
Update youtube shortcode to accept timestamp

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Current `youtube` shortcode offers no option to set the start time for an embedded youtube video.

Why? Some videos from the [accessibility-for-teams](https://github.com/GSA/digitalgov.gov/pull/7279) guide migration use specific start times to accompany text content.
It would be useful to keep these video bookmarks after the migration and it will also provide the functionality in the future if needed.

### Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-youtube-shortcode/event/2018/09/27/a-conversation-with-evan-papp/)

### Solution

This enhancement adds two fields for setting the required `id` and optional `time`.

**Before**

```
{{< youtube Hvo7WVKvPGU >}}
```

**After**
```
{{< youtube id="Hvo7WVKvPGU" title="Youtube video title" start_time="101" >}}
```

The `title` field sets the html `title` attribute on the `iframe` to meet accessibility.

### Checklist

- [x] updated shortcode partial
- [x] updated existing shortcode instances
- [x] documented enhancements in code
- [ ] updated wiki

### How To Test

1. Visit [page with youtube embed](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-youtube-shortcode/event/2018/09/27/a-conversation-with-evan-papp/) (the second video on the page)
2. Add `start_time` to shortcode on local build
3. Visit page, press play and video should start at provided time

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

**Comment by nick-mon1** [comment]
@bonnieAcameron I've added some comments for files that have youtube embeds that have been deleted. Please take a look if offer any language changes for any above/below text content that references the now deleted video.

- content/news/2013/02/2013-02-11-youll-have-them-at-swipe-making-an-awesome-mobile-user-experience-webinar.md
- content/news/2013/02/2013-02-14-qr-codes.md
- content/news/2016/03/2016-03-08-quizzes-the-secret-weapon-of-the-worlds-best-content-creators.md
- content/news/2018/04/2018-04-02-marketing-government-through-agency-partnerships.md
- content/news/2016/07/2016-07-14-analytics-trainings-2016-so-far.md
- content/news/2016/05/2016-05-12-using-pirate-metrics-to-analyze-your-mobile-applications-audience.md
- content/news/2017/09/2017-09-20-a-conversation-with-itif-about-state-federal-government-websites.md
- content/resources/digitalgov-user-experience-program-speaking-opportunities.md
- content/resources/digitalgov-user-experience-program-speaking-opportunities.md
- content/resources/inventory-data-gov-guide.md

**Review by bonnieAcameron — DISMISSED** [review]

File: themes/digital.gov/layouts/shortcodes/youtube.html
Comment (ToniBonittoGSA) [inline themes/digital.gov/layouts/shortcodes/youtube.html:1]: Should

> @example
  {{< youtube id="Hvo7WVKvPGU" time"101" title="Title goes here">}}

change to

> @example
  {{< youtube id="Hvo7WVKvPGU" time="101" title="Title goes here" >}}

Comment (nick-mon1) [reply themes/digital.gov/layouts/shortcodes/youtube.html:1]: Thanks @ToniBonittoGSA, I've updated the example documentation.

File: themes/digital.gov/layouts/shortcodes/youtube.html [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+{{ $title := (.Get "title") }}
+
+
 <div
   class="video"
   style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;"
 >
   <iframe
-    src="//www.youtube.com/embed/{{ index .Params 0 }}"
+    src="https://www.youtube.com/embed/{{ $id }}?start={{ if $start_time }}
```
Comment (mejiaj) [inline themes/digital.gov/layouts/shortcodes/youtube.html:23]: Does the `?start=` still appear if you don't enter a value for `$start_time`?

Comment (nick-mon1) [reply themes/digital.gov/layouts/shortcodes/youtube.html]: It does, moved the conditional to only show when `$start_time` exists. Disabled prettier formatting which was causing white spaces with the conditional check for the `$start_time`, see old formatting (02ce5c474a82ae28ec1e294fb4ad0e8a8209033d)

Comment (mejiaj) [reply themes/digital.gov/layouts/shortcodes/youtube.html]: What was the prettier issue? We should avoid ignoring wherever possible.

**Review by mejiaj — APPROVED** [review]
Confirming that the partial works as expected and comment has been addressed.

We should capture the following in Trello (if we haven't already):

- [ ] Follow-up for deleted videos
- [ ] Avoid `prettier-ignore`

**Comment by nick-mon1** [comment]
@mejiaj Created trello card for the suggested follow-up items. ✅

**Pull request merged**